sentry-android-core/src/main/java/io/sentry/android/core/EnvelopeFileObserverIntegration.java

```diff
@@ -37,28 +37,42 @@ public final void register(final @NotNull IHub hub, final @NotNull SentryOptions
       logger.log(
           SentryLevel.DEBUG, "Registering EnvelopeFileObserverIntegration for path: %s", path);
 
-      final OutboxSender outboxSender =
-          new OutboxSender(
-              hub,
-              options.getEnvelopeReader(),
-              options.getSerializer(),
-              logger,
-              options.getFlushTimeoutMillis());
-
-      observer =
-          new EnvelopeFileObserver(path, outboxSender, logger, options.getFlushTimeoutMillis());
       try {
-        observer.startWatching();
-        logger.log(SentryLevel.DEBUG, "EnvelopeFileObserverIntegration installed.");
+        options.getExecutorService().submit(() -> startOutboxSender(hub, options, path));
       } catch (Throwable e) {
-        // it could throw eg NoSuchFileException or NullPointerException
-        options
-            .getLogger()
-            .log(SentryLevel.ERROR, "Failed to initialize EnvelopeFileObserverIntegration.", e);
+        logger.log(
+            SentryLevel.DEBUG,
+            "Failed to start EnvelopeFileObserverIntegration on executor thread. Starting on the calling thread.",
+            e);
+        startOutboxSender(hub, options, path);
       }
     }
   }
 
+  private void startOutboxSender(
```

Comment:
`m`: Probably makes sense to synchronize this method and `.close()` to ensure no race conditions when accessing the observer field.
We probably even need a `isClosed` field in case the executor service is running `startOutboxSender()` after `close()` is called.